tools/Shared/Microsoft.TemplateEngine.CommandUtils/DotnetCommand.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Logging;
+using Xunit.Abstractions;
+
+namespace Microsoft.TemplateEngine.CommandUtils
+{
+    internal class DotnetCommand : TestCommand
+    {
+        internal DotnetCommand(ILogger log, string subcommand, params string[] args) : base(log)
+        {
+            Arguments.Add(subcommand);
+            Arguments.AddRange(args);
+            //opt out from telemetry
+            WithEnvironmentVariable("DOTNET_CLI_TELEMETRY_OPTOUT", "true");
```

Comment:
We definitely want to exclude telemetry for our CIs. I'm not sure how we want to look on the tests executed by 3rd parties via `TemplateVerifier` in their environments... But it's probably best to exclude everything - as then we'd need to allow distinguishing between 1st party and 3rd party usages of `TemplateVerifier`... and test invocations are not actual instantiations anyways.

@baronfel as FYI - there is a slight (and hopefully unprobable) effect on our telemetry stats - in case there would be  3rd parties that are executing some custom tests today (or even 1st parties that do not properly exclude telemetry) and would switch to `TemplateVerifier` - than we might see a declines in stats that are not real.

Reply:
Maybe it would be better to let the caller to decide on this?
But then we need to extra work in all the tests.

Reply:
On of the TODOs for the Templates Verification FW is addition of telemetry (tailored to usage of the authoring tooling) - that way (if we exclude that one as well in our integration tests) - we'd be able to get the idea of how much it's been used by other authors - and hence estimate the diff gap caused by excluding the dotnet CLI telemetry.

Reply:
After discussion we agreed to make telemetry output optional. @JanKrivanek already has a plan to implement bypassing env variables in test framework shortly.